stan/math/opencl/prim/cholesky_decompose.hpp

```diff
@@ -29,8 +30,11 @@ inline matrix_cl<double> cholesky_decompose(const matrix_cl<double>& A) {
   }
   opencl::cholesky_decompose(res);
   // check_pos_definite on matrix_cl is check_nan + check_diagonal_zeros
-  check_nan("cholesky_decompose (OpenCL)", "A", res);
-  check_diagonal_zeros("cholesky_decompose (OpenCL)", "A", res);
+  check_cl("cholesky_decompose (OpenCL)", "A", res, "not NaN") = !isnan(res);
```

Comment:
Should this check that it is NaN?

Reply:
Check is requiring the values not to be nan in order for check to pass. That is in line with what this previously did, right?

Reply:
Yeah. Then I guess the next one should check that the diagonal is not zero.

Reply:
However I did the other check backwards. Fixed.